drivers/hmis/app/graphql/mutations/delete_form_definition.rb

```diff
@@ -0,0 +1,27 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class DeleteFormDefinition < CleanBaseMutation
+    argument :id, ID, required: true
+
+    field :form_definition, Types::Forms::FormDefinition, null: true
+
+    def resolve(id:)
+      raise 'not allowed' unless current_user.can_configure_data_collection?
+
+      definition = Hmis::Form::Definition.find_by(id: id)
+      raise 'not found' unless definition
+
+      definition.destroy!
```

Comment:
Can we add a check here to disallow deleting definitions that are referenced by any rules? This should just help ensure that things arent deleted by accident

```
raise 'can't delete definition with active rules' if definition.instances.active.exists?
```